src/PolyType/Abstractions/IEnumTypeShape.cs

```diff
@@ -25,4 +25,14 @@ public interface IEnumTypeShape<TEnum, TUnderlying> : ITypeShape<TEnum>, IEnumTy
     /// Gets the shape of the underlying type used to represent the enum.
     /// </summary>
     new ITypeShape<TUnderlying> UnderlyingType { get; }
+
+    /// <summary>
+    /// Gets the names and values for each enum member.
+    /// </summary>
+    /// <devremarks>
+    /// We use <typeparamref name="TUnderlying"/> instead of <typeparamref name="TEnum"/>
+    /// to avoid needlessly closing generic types around unique value types (which swell the size of trimmed apps)
+    /// when the caller can easily cast between enum and underlying types.
+    /// </devremarks>
+    IReadOnlyDictionary<string, TUnderlying> Members { get; }
```

Comment:
Using `string` as the key instead of `TUnderlying` is because the name of the member is guaranteed to be unique whereas the value is _not_ guaranteed to be unique.

I'm not adding Func's that convert between the two as @eiriktsarpalis had proposed in the issue because it assumes converting between string or `ReadOnlySpan<char>` is useful, which in msgpack it will not be. And whatever _is_ useful to the PolyType user can be built up all based on this dictionary.

Reply:
Is this dictionary meant to function as a substitute for `Enum.Parse` with customized names? I don't think the current implementation accounts for composite values in `[Flags]` enums. The built-in enum converter in STJ does do this [here](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs) in case you're looking for a reference implementation.

Reply:
I don't see how composite values are necessarily relevant. The names are unique and each have a value assigned, even if the value is not unique. Yes, during deserialization if you encounter a name, you can translate that to a number using this dictionary. Arbitrary combinations of flags (if the enum is flags) is the responsibility of the serializer -- not this map, IMO.

Reply:
> Arbitrary combinations of flags (if the enum is flags) is the responsibility of the serializer -- not this map, IMO.

You're possibly right, although I would argue that handling combinations this way is common enough and complex enough to warrant a centralized solution.

Reply:
Given it _can_ be added on top of this, I'd prefer to leave it out, at least for now. 
If serialization is enough of a focus for PolyType to merit specialized code for it, and if it can add value by providing helper methods for recognizing how a value is composed of several named values _without_ making any assumptions about how those multiple names will be strung together (e.g. commas, spaces, pipes, etc) then adding it (later?) seems fine to me. 
For my part though, I'm rather content to emit a single name, or an ordinal value in msgpack

Reply:
Regarding delegates, I want to have access to the whole dictionary so I can enumerate possibilities and create my own lookup table based on UTF8 encoded characters.

Reply:
Makes sense.

Reply:
Okay I'll give it a shot.

Reply:
No, I don't think so. All of these are based on strings and UTF-16 characters. MessagePack is based on UTF-8. The `IsDefinedValueOrCombinationOfValues` method is independent of character encoding, but it only returns a `bool` which is of no use to me.